psyneulink/core/llvm/helpers.py

```diff
@@ -239,6 +228,42 @@ def is_boolean(x):
         type_t = x.type.pointee
     return isinstance(type_t, ir.IntType) and type_t.width == 1
 
+def get_array_shape(x):
+    x_ty = x.type
+    if is_pointer(x):
+        x_ty = x_ty.pointee
+
+    assert isinstance(x_ty, ir.ArrayType), f"Tried to get shape of non-array type: {x_ty}"
+    dimensions = []
+    while hasattr(x_ty, "count"):
+        dimensions.append(x_ty.count)
+        x_ty = x_ty.element
+
+    return tuple(dimensions)
```

Comment:
Why does this need to return tuple?

Reply:
this is to match the type of numpy's shape attribute, which is a tuple. This probably isn't strictly necessary, but I didn't see a reason to allow for mutable shapes and to deviate from numpy's type.

Reply:
I don't think matching python in anything but computational semantics should be a priority.

Reply:
is the fix here to return the list without casting?

Reply:
yeah, unless something depends on it being a tuple. it might be easier to pass it as an initializer (ctypes likes tuples, dunno about llvmlite). that was the original question.
If the only reason is to stay close to Python, we can drop the cast.